java/flight/flight-core/src/main/java/org/apache/arrow/flight/ArrowMessage.java

```diff
@@ -386,8 +388,9 @@ public ArrowMessage parse(InputStream stream) {
   }
 
   @Override
-  public void close() throws Exception {
-    AutoCloseables.close(bufs);
+  public void close() {
+    // Don't use AutoCloseables to avoid a checked exception
+    bufs.forEach(ArrowBuf::close);
```

Comment:
This isn't safe since foreach will fail the moment you hit the first exception. You need to make sure that even if an exception occurs, the rest of the items get closed (and then append suppressed exceptions). That's what AutoCloseables.close does.

Reply:
I changed this to use AutoCloseables again, and to propagate the exception.